### DISCUSSION TITLE
Node install on Windows Production Best Practices?

### DISCUSSION THREAD

**Discussion by RayDeanTech** [coreybutler/nvm-windows#578] [Support]
I'm willing to bet money that `NVM_HOME` is the path at which you installed nvm-windows.  But what is the intended purpose for `NVM_SYMLINK`?

Is there anything wrong with putting _everything_ under `C:\nodejs` ?

So, like, I want everything under the C:\nodejs.  Does this make sense for a Production Node.js deployment to Windows?

`C:\nodejs`
`C:\nodejs\nvm`
`C:\nodejs\node.exe`
`C:\nodejs\node_modules`
`C:\nodejs\node_modules\npm\`
`C:\nodejs\node_modules\some-global-install\`  (think PM2)
`C:\nodejs\npm-cache`

**Comment by RayDeanTech** [comment]
Now that I've read some more, I personally think this feels more like it's for DEV vs PROD.  I think in PROD, I'm not going to be switching node versions that often.

Still would love to see opinions on install directories, though! :)

**Comment by coreybutler** [comment]
For production, I tend to use containers. In dev, it's fine to put things under `C:\nodejs`.

`NVM_SYMLINK` is the _virtual representation_ of the Node installation directory, i.e. a redirect. Windows searches the `PATH` to find a program when you run it (i.e. running `node index.js` is auto-resolved to `C:\path\to\node.exe index.js`). By adding a symlink to the `PATH`, Windows will automatically find the appropriate directory because it will follow the symlink to its destination. Every time you run `nvm use x.x.x`, it updates the _target_ of the symlink specified in `NVM_SYMLINK`.
